crates/data/src/lib.rs

```diff
@@ -0,0 +1,66 @@
+//! Common data definitions for sightglass.
+//!
+//! These are in one place, pulled out from the rest of the crates, so that many
+//! different crates can serialize and deserialize data by using the same
+//! definitions.
+
+#![deny(missing_docs, missing_debug_implementations)]
+
+use serde::{Deserialize, Serialize};
+use std::borrow::Cow;
+
+/// A single measurement, for example instructions retired when compiling a Wasm
+/// module.
+///
+/// This is often used with the `'static` lifetime when recording measurements,
+/// where we can use string literals for various fields. When reading data, it
+/// can be used with a non-static lifetime to avoid many small allocations.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Measurement<'a> {
+    /// The CPU architecture on which this measurement was taken, for example
+    /// "aarch64" or "x86_64".
+    pub arch: Cow<'a, str>,
+
+    /// The file path of the wasmtime benchmark API shared library used to
+    /// record this measurement.
+    pub engine: Cow<'a, str>,
+
+    /// The file path of the Wasm benchmark program.
+    pub wasm: Cow<'a, str>,
+
+    /// The id of the process within which this measurement was taken.
+    pub process: u32,
```

Comment:
I would call this `process_id` just to make it more clear for humans reading the raw CSV.